### PR TITLE
:bug: Adding ability for the Maven object to create the settings file

### PR DESCRIPTION
**Pull request by shawn-hurley**
We are also adding ability to set the local repository to the M2Dir.

### CODE REVIEW

File: repository/maven.go [hunk 1]
```diff
@@ -155,24 +152,42 @@ func (r *Maven) writeSettings() (path string, err error) {
...
 			"path",
 			path)
 	}
-	_ = f.Close()
 	addon.Activity("[FILE] Created %s.", path)
 	return
 }
 
-//
+func (r *Maven) injectCacheDir(settings string) (string, error) {
```
Comment (shawn-hurley) [inline repository/maven.go:172]: If we are going to do chaining more in the future, a way to pass around the mxj object might be worth exploring

File: repository/maven.go [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
 // Maven repository.
 type Maven struct {
 	Remote
 	BinDir string
 	M2Dir  string
 }
 
-//
+func (r *Maven) CreateSettingsFile() (string, error) {
+	return r.writeSettings()
```
Comment (jortel) [inline repository/maven.go:22]: Let's move the content of writeSettings() here.  Let's change the signature to pass in the path for better utility.
CreateSettingsFile(path string) (err error)
With the following flow:

var settings string

1. Get the identity.
if found, settings = id.Settings else settings = _EmptySettings_
2. m, err = parse XML
3.err =  r.injectCacheDir(m *mxj.Map)
4. err = r.injectProxies(m *mxj.Map)
5. write file.

Comment (jortel) [reply repository/maven.go:22]: EmptySettings would a _const_ with something like:
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.2.0" 
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
                 xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0
                 http://maven.apache.org/xsd/settings-1.2.0.xsd">
</settings>
```

File: repository/maven.go (hunk 1 above)
Comment (shawn-hurley) [reply repository/maven.go]: updated

File: repository/maven.go (hunk 2 above)
Comment (shawn-hurley) [reply repository/maven.go:30]: updated

**Comment by shawn-hurley** [comment]
@jortel @mansam This is ready for review, please let me know if there is something that you need changed

File: repository/maven.go [hunk 3]
```diff
@@ -155,35 +148,66 @@ func (r *Maven) writeSettings() (path string, err error) {
...
+	_, err = f.Write(settings)
 	if err != nil {
 		err = liberr.Wrap(
 			err,
 			"path",
 			path)
 	}
-	_ = f.Close()
-	addon.Activity("[FILE] Created %s.", path)
+	addon.Activity("[FILE] Created %s. -- file: %s", path, settings)
```
Comment (jortel) [inline repository/maven.go:188]: We cannot include the settings file in the task activity log because it contains credentials.

File: repository/maven.go [hunk 4]
```diff
@@ -155,35 +148,66 @@ func (r *Maven) writeSettings() (path string, err error) {
...
+		return nil, err
+	}
+	return settings, nil
+}
+
 // injectProxy injects proxy settings.
-func (r *Maven) injectProxy(id *api.Identity) (s string, err error) {
-	s = id.Settings
-	m, err := mxj.NewMapXml([]byte(id.Settings))
+func (r *Maven) injectProxy(id *api.Identity) (mxj.Map, error) {
```
Comment (jortel) [inline repository/maven.go:202]: All of the core projects use named return values by convention. Why did you change this?
Also, my previous comment requested the injectProxy() take a mxj.Map to be consistent with injectCacheDir() and because both the m2 dir and proxy settings should be set on either the emptySettings or the setting in the Identity.  Right?

**Comment by jortel** [comment]
Last I checked, ubi9 contains Go 1.18 so we have been development with it to ensure compatibility.  I assume the widespread reformatting was because you used gomft 1.19+ which has, imho, become rather draconian.  If so, that's fine, all of the files will get reformatted when the project updates Go. If not, please revert the unrelated changes.

**Review by jortel — APPROVED** [review]
LGTM
Thanks @shawn-hurley !
